src/Router.js

```diff
@@ -85,62 +70,119 @@ export default class Router {
       throw new Error('Router#registerRoutes must be passed an array of Routes')
     }
     routes.map(route => {
-      this.__routes.push(new Route(route))
+      route = new Route(route);
+      if (this.onRouteComplete) {
+        const handlerLength = route.handlers.length-1;
+        const lastHandler = route.handlers[handlerLength];
+        const routingEnd = (ctx, next) => {
```

Comment:
This was super hacky before because `page.js` expects the last function to not have a `next()` function.

Instead it'd be nicer to `push` the `config.onRouteComplete` function if it is present and not have to hackily slice up the array of handler functions